enterprise/backend/src/metabase_enterprise/gsheets.clj

```diff
@@ -0,0 +1,351 @@
+             :as   gdrive-conn} (normalize-gdrive-conn conn)
+            last-dwh-sync       (get-last-mb-dwh-sync-time)]
+        (-> (cond
+              (sync-complete? {:status status :last-dwh-sync last-dwh-sync :last-gdrive-conn-sync last-gdrive-conn-sync})
+              (u/prog1 (assoc (gsheets) :status "complete")
+                (gsheets! <>)
+                (snowplow/track-event! ::snowplow/simple_event
+                                       {:event "sheets_connected" :event_detail "success"}))
+
+              (when-let [upload-time (:folder-upload-time (gsheets))]
+                (> (seconds-from-epoch-now) (+ upload-time *folder-setup-timeout-seconds*)))
+              (do (reset-gsheets-status!)
+                  (error-response-in-body (tru "Timeout syncing google drive folder, please try again.")
+                                          {:status-code 408}))
+
+              ;; Syncing failed
+              (= "error" status)
+              (do
+                (reset-gsheets-status!)
+                (error-response-in-body (tru "Problem syncing google drive folder, please try again..")))
+
+              ;; Continue waiting
+              :else (gsheets))
+            (assoc :db_id (:id attached-dwh)
+                   ;; TEMP (gsheets)
+                   ;; here is some debugging info to make sure we have it straight:
+                   :hm/conn gdrive-conn
+                   :mb/sync-info {:status                status
+                                  :last-dwh-sync         last-dwh-sync
+                                  :last-gdrive-conn-sync last-gdrive-conn-sync})))
+      (do
+        (reset-gsheets-status!)
+        (error-response-in-body
+         (tru "Unable to find google drive connection.")
+         {:status-code 404})))))
+
+(api.macros/defendpoint :get "/folder" :- ::gsheets
+  "Check the status of a newly created gsheets folder creation. This endpoint gets polled by FE to determine when to
+  stop showing the setup widget.
+
+  Returns the gsheets shape, with the attached datawarehouse db id at `:db_id`."
+  [] :- ::gsheets
+  (api/check-superuser)
+  (let [attached-dwh (t2/select-one :model/Database :is_attached_dwh true)]
+    (when-not (some? attached-dwh)
+      (snowplow/track-event! ::snowplow/simple_event {:event "sheets_connected" :event_detail "fail - no dwh"})
+      (reset-gsheets-status!)
+      (error-response-in-body (tru "No attached dwh found.")))
+    (handle-get-folder attached-dwh)))
+
+(api.macros/defendpoint :delete "/folder"
+  "Disconnect the google service account. There is only one (or zero) at the time of writing."
+  []
+  (api/check-superuser)
+  (snowplow/track-event! ::snowplow/simple_event {:event "sheets_disconnected"})
+  (reset-gsheets-status!))
+
+(def ^{:arglists '([request respond raise])} routes
+  "`/api/ee/gsheets` routes."
+  ;; TODO add api/+check-superuser +auth and +check-setting
+  (api.macros/ns-handler *ns*))
+
+(comment
```

Comment:
want to remove this or are you including intentionally?

Reply:
fixed